internal/pkg/cli/pipeline_init.go

```diff
@@ -192,6 +201,49 @@ func (o *initPipelineOpts) Ask() error {
 	return nil
 }
 
+func (o *initPipelineOpts) validateDuplicatePipeline() error {
+	// make sure pipeline isn't already deployed
+	fullName := fmt.Sprintf(fmtPipelineName, o.appName, o.name)
+
+	_, err := o.pipeline.GetPipeline(fullName)
```

Comment:
Just curious to know if there is any priority order for calling `o.pipeline.GetPipeline(fullName)` vs `o.wsLister.ListPipelines()`? If we can do local check 1st and can eliminate the error case then only we can make an API call to check further is what my understanding is. Please correct me if I am missing something.

Reply:
Yeah - It would be nice to avoid the unnecessary API call. However, my thought was that if we did the local check (`ListPipelines()`) before the remote check (`GetPipeline()`) and they put in a name that has already been deployed, we'd see the local conflict but still need to do the remote check in order to print out the correct error message (whether they need to `delete` the pipeline or if they have yet to `deploy` it)

Does that thinking make sense? Or am I crazy lol